app/src/main/java/io/gnosis/safe/ui/transaction/TransactionViewHolderFactory.kt

```diff
@@ -38,64 +41,81 @@ class TransactionViewHolderFactory : BaseFactory<BaseTransactionViewHolder<Trans
+            is TransactionView.SettingsChangeQueued -> TransactionViewType.SETTINGS_CHANGE_QUEUED
+            is TransactionView.TransferQueued -> TransactionViewType.TRANSFER_QUEUED
             else -> throw UnsupportedViewType(javaClass.name)
         }.ordinal
 }
 
-abstract class BaseTransactionViewHolder<T : Transaction>(viewBinding: ViewBinding) : Adapter.ViewHolder<T>(viewBinding.root)
+
+abstract class BaseTransactionViewHolder<T : TransactionView>(viewBinding: ViewBinding) : Adapter.ViewHolder<T>(viewBinding.root)
 
 class ChangeMastercopyViewHolder(viewBinding: ItemTxChangeMastercopyBinding) :
-    BaseTransactionViewHolder<Transaction.ChangeMastercopy>(viewBinding) {
+    BaseTransactionViewHolder<TransactionView.ChangeMastercopy>(viewBinding) {
 
-    override fun bind(data: Transaction.ChangeMastercopy, payloads: List<Any>) {
+    override fun bind(data: TransactionView.ChangeMastercopy, payloads: List<Any>) {
         TODO("Not yet implemented")
     }
 }
 
-class SettingsChangeViewHolder(viewBinding: ItemTxSettingsChangeBinding) :
-    BaseTransactionViewHolder<Transaction.SettingsChange>(viewBinding) {
+class SettingsChangeViewHolder(private val viewBinding: ItemTxSettingsChangeBinding) :
+    BaseTransactionViewHolder<TransactionView.SettingsChange>(viewBinding) {
 
-    override fun bind(data: Transaction.SettingsChange, payloads: List<Any>) {
-        TODO("Not yet implemented")
+    override fun bind(data: TransactionView.SettingsChange, payloads: List<Any>) {
+        with(viewBinding) {
+            settingName.text = data.transaction.dataDecoded.method
+            dateTime.text = data.transaction.date
+        }
     }
 }
 
-class TransferViewHolder(viewBinding: ItemTxTransferBinding) :
-    BaseTransactionViewHolder<Transaction.Transfer>(viewBinding) {
-
-    override fun bind(data: Transaction.Transfer, payloads: List<Any>) {
-        TODO("Not yet implemented")
+class TransferViewHolder(private val viewBinding: ItemTxTransferBinding) :
+    BaseTransactionViewHolder<TransactionView.Transfer>(viewBinding) {
+
+    override fun bind(data: TransactionView.Transfer, payloads: List<Any>) {
+        with(viewBinding) {
+            val value = data.transaction.value
+            amount.text = "${value.shiftedString(18)} ${data.transaction.tokenInfo?.symbol}"
+            dateTime.text = data.transaction.date
+            if (data.isIncoming) {
+                txTypeIcon.setImageResource(R.drawable.ic_arrow_green_16dp)
+                blockies.setAddress(data.transaction.sender)
+                ellipsizedAddress.text = data.transaction.sender.asEthereumAddressString().asMiddleEllipsized(4)
+            } else {
+                txTypeIcon.setImageResource(R.drawable.ic_arrow_red_10dp)
+                blockies.setAddress(data.transaction.recipient)
+                ellipsizedAddress.text = data.transaction.recipient.asEthereumAddressString().asMiddleEllipsized(4)
```

Comment:
maybe it is a good idea to extract this ethereum address formatting to a utility method, so that formatting can be edited in one place

Reply:
I extracted it as `formatForTxList()`. But where should I put it?

Reply:
io.gnosis.safe.utils.StringUtils?

Reply:
Fixed in: d43eacfa613733d0eb13f09c533224ce0a05a8d9